### ISSUE TITLE
Pyright generates error "unpacking not allowed" in context of __get_item__calls for numpy, tensorflow tensors

### ISSUE THREAD

**Issue by dmlyubim** [microsoft/pyright#3068]
Note: if you are reporting a wrong signature of a function or a class in the standard library, then the typeshed tracker is better suited for this report: https://github.com/python/typeshed/issues.

**Describe the bug**
"unpack operation is not allowed in this context" is generated. However, in python 3.8 this executes just fine. 

**To Reproduce**

>>>
import numpy as np
a = np.array([[1,2]])
b = a[ (*[np.newaxis]*3), ...]    # <<-- generates pyright error
b

array([[[[[1, 2]]]]])

**Expected behavior**
no pyright error 

**Screenshots or Code**

**Additional context**
python 3.8, current versions of numpy and/or tensorflow, VSCode pylance plugin

**Comment by erictraut** [comment]
It looks like this syntax form was deprecated in Python 3.9. When I run this sample in the Python interpreter, I receive a runtime exception. It does work OK in Python 3.8.

Here's what I see on Python 3.9.
```
    b = a[ (*[np.newaxis]*3), ...]
            ^
SyntaxError: can't use starred expression here
```

And in Python 3.10:
```python
    b = a[ (*[np.newaxis]*3), ...] # <<-- generates pyright error
            ^^^^^^^^^^^^^^^
SyntaxError: cannot use starred expression here
```

Since this syntax form is deprecated, it's probably best to avoid using it.

**Comment by dmlyubim** [comment]
i don't know of alternatives to insert n new dimensions (or skip over n slices) when reforming tensors. let me check what exists for 3.9

**Comment by dmlyubim** [comment]
for 3.9, nested list seems to work, something like a[[ *[np.newaxis]*3, ... ]]. And pyright is ok with this.
